tests/common/plugins/sanity_check/checks.py

```diff
@@ -176,6 +246,7 @@ def check_processes(dut):
         start = time.time()
         elapsed = 0
         while elapsed < timeout:
+            check_result["failed"] = False
```

Comment:
@yxieca Please also check this change.

Reply:
What specifically you want me to check? Please elaborate.

Reply:
Let me explain why I did this change although this change was not relevant to sanity check of Monit service. I found this bug when doing the test of sanity check for Monit service. 

Suppose some critical processes was not running when doing the sanity check, then at the first iteration, `failed` field in `check_result` will be marked as `True` at line 184. Next iteration when all the critical processes become live, the line 184 will be skipped. However the value of `check_result["failed"]` was still `True`. This will cause the re-execution of line 187 ~ 190 and then do the iteration again if it is not timeout.  Since the value of `check_result["failed"]` was still be `True`, sanity check will finally fail.

Please help me verify whether such change is correct or not.